### PR TITLE
Time based whitelists

### PR DESCRIPTION
**Pull request by dakaneye**
Signed-off-by: Samuel Dacanay <sam.dacanay@anchore.com>

**What this PR does / why we need it**: Adds the ability for a whitelist to expire

**Which issue this PR fixes** Fixes https://github.com/anchore/enterprise/issues/369

**Special notes**:

### CODE REVIEW

**Comment by dakaneye** [comment]
@zhill I did an initial pass at this to see if this is heading in the right direction.

**Comment by dakaneye** [comment]
Just verified locally, able to get whitelist checks to expire!

**Comment by dakaneye** [comment]
I've updated this PR now to set an expiry on each whitelist item (rule) instead of the entire whitelist set.

File: tests/data/bundle-large_whitelist.json [hunk 1]
```diff
@@ -13470,6 +13470,20 @@
       ],
       "name": "Global Whitelist",
       "version": "1_0"
+    },
+    {
```
Comment (zhill) [inline tests/data/bundle-large_whitelist.json:13474]: Great that this data is here, but is this used in an actual test?

**Review by zhill — CHANGES_REQUESTED** [review]
Overall looks pretty good @dakaneye ! I'd like an explicit test (unit or integration) for expired rules, but otherwise just a small non-blocking suggestion.

File: anchore_engine/services/policy_engine/engine/policy/bundles.py [hunk 2]
```diff
@@ -836,6 +842,13 @@ def execute(self, trigger_match):
         :param trigger_inst: the trigger instance to check
         :return: a WhitelistedTriggerInstance or a TriggerInstance depending on if the items match
         """
+        # If this whitelist rule is expired, we do not search for any matches
+        # When expires_on is parsed it's translated to UTC, so we must do the same for getting the current time
+        # so that we can compare them
+        now_utc = anchore_now_datetime().replace(tzinfo=datetime.timezone.utc)
```
[the remark this answers is not in the document]
Comment (dakaneye) [reply anchore_engine/services/policy_engine/engine/policy/bundles.py]: hoisted! and added a unit test

File: tests/data/bundle-large_whitelist.json (hunk 1 above)
Comment (dakaneye) [reply tests/data/bundle-large_whitelist.json:13474]: as discussed, used for implicit schema validation and execution

File: anchore_engine/services/apiext/swagger/swagger.yaml [hunk 3]
```diff
@@ -2933,6 +2933,9 @@ definitions:
         type: string
       trigger_id:
         type: string
+      expires_on:
```
[the remark this answers is not in the document]
Comment (dakaneye) [reply anchore_engine/services/apiext/swagger/swagger.yaml:2973]: 👍

**Review by zhill — APPROVED** [review]
lgtm

**Pull request merged**